test/srtp_driver.c

```diff
@@ -4287,3 +4299,240 @@ const srtp_policy_t wildcard_policy = {
     0,    /* list of encrypted extension headers is empty */
     NULL
 };
+
+static srtp_stream_t stream_list_test_create_stream(uint32_t ssrc)
+{
+    srtp_stream_t stream = calloc(1, sizeof(srtp_stream_ctx_t));
+    stream->ssrc = ssrc;
+    return stream;
+}
+
+static void stream_list_test_free_stream(srtp_stream_t stream)
+{
+    free(stream);
+}
+
+int stream_list_test_count_cb(srtp_stream_t stream, void *data)
+{
+    int *count = (int *)data;
+    (*count)++;
+    (void)stream;
+    return 0;
+}
+
+struct remove_one_data {
+    uint32_t ssrc;
+    srtp_stream_list_t list;
+};
+
+int stream_list_test_remove_one_cb(srtp_stream_t stream, void *data)
+{
+    struct remove_one_data *d = (struct remove_one_data *)data;
+    if (stream->ssrc == d->ssrc) {
+        srtp_stream_list_remove(d->list, stream);
+        stream_list_test_free_stream(stream);
+        return 1;
+    }
+    return 0;
+}
+
+int stream_list_test_remove_all_cb(srtp_stream_t stream, void *data)
+{
+    srtp_stream_list_t *list = (srtp_stream_list_t *)data;
+    srtp_stream_list_remove(*list, stream);
+    stream_list_test_free_stream(stream);
+    return 0;
+}
+
+srtp_err_status_t srtp_stream_list_test(void)
+{
+    srtp_stream_list_t list;
+
+    if (srtp_stream_list_alloc(&list)) {
+        return srtp_err_status_fail;
+    }
+
+    /* add 4 streams*/
+    if (srtp_stream_list_insert(list, stream_list_test_create_stream(1))) {
+        return srtp_err_status_fail;
+    }
+    if (srtp_stream_list_insert(list, stream_list_test_create_stream(2))) {
+        return srtp_err_status_fail;
+    }
+    if (srtp_stream_list_insert(list, stream_list_test_create_stream(3))) {
+        return srtp_err_status_fail;
+    }
+    if (srtp_stream_list_insert(list, stream_list_test_create_stream(4))) {
+        return srtp_err_status_fail;
+    }
+
+    /* find */
```

Comment:
Would be good to test for a false positive as well, say `srtp_stream_list_get(list, 5) != NULL`

Reply:
see the comment a bit further down in the file:  /* find not in list */
that is I guess what you meant ?

Reply:
Yep, thanks.